yt/frontends/amrex/fields.py

```diff
@@ -471,43 +473,27 @@ def setup_fluid_fields(self):
         for _, field in self.ds.field_list:
             if field.startswith("X("):
                 # We have a mass fraction
-                nice_name, tex_label = _nice_species_name(field)
+                sub = Substance(field)
                 # Overwrite field to use nicer tex_label display_name
                 self.add_output_field(
                     ("boxlib", field),
                     sampling_type="cell",
                     units="",
-                    display_name=tex_label,
-                )
-                self.alias(
-                    ("gas", f"{nice_name}_fraction"), ("boxlib", field), units=""
+                    display_name=sub.to_tex(),
                 )
-                func = _create_density_func(("gas", f"{nice_name}_fraction"))
+                self.alias(("gas", f"{sub}_fraction"), ("boxlib", field), units="")
+                func = _create_density_func(("gas", f"{sub}_fraction"))
                 self.add_field(
-                    name=("gas", f"{nice_name}_density"),
+                    name=("gas", f"{sub}_density"),
                     sampling_type="cell",
                     function=func,
                     units=unit_system["density"],
-                    display_name=rf"\rho {tex_label}",
+                    display_name=rf"\rho {sub.to_tex()}",
                 )
 
-                # Most of the time our species will be of the form
-                # element name + atomic weight (e.g. C12), but
-                # sometimes we make up descriptive names (e.g. ash)
-                if any(char.isdigit() for char in field):
-                    # We know this will either have one letter, or two.
-                    if field[3] in string.ascii_letters:
-                        element, weight = field[2:4], field[4:-1]
-                    else:
-                        element, weight = field[2:3], field[3:-1]  # NOQA
-                    weight = int(weight)
-
-                # Here we can, later, add number density using 'element' and
-                # 'weight' inferred above
```

Comment:
As I was refactoring this branch, I realized it had really been left in a noop "TODO" state for 10 years and I feel like it's not worth keeping it around given that it creates discussion around code that's actually not used.

Reply:
We should also remove the same code in `CastroFieldInfo.setup_fluid_fields()`.

Reply:
Good point. Actually a separable problem so I'll open another PR. Thank you !